tests/unit/test_logging.py

```diff
@@ -0,0 +1,224 @@
+        except json.JSONDecodeError as err:
+            raise RuntimeError("Log doesn't seem to be in JSON format") from err
+
+    return capture_log
+
+
+def test_configured_log_level(caplog, expect_json_log):
+    """Test with config"""
+    logger = logging.getLogger("test_configured_log_level")
+    config = LoggingConfig(
+        log_level="CRITICAL",  # set to higher than default of WARNING
+        service_name=VALID_SERVICE_NAME,
+        service_instance_id=VALID_INSTANCE_ID,
+        log_format="",
+    )
+
+    configure_logging(logger=logger, config=config)
+
+    assert not caplog.records
+
+    # assert nothing was captured due to config
+    logger.warning("should not log")
+    assert not caplog.records
+
+    # verify that the log is emitted and that it's the one we expect
+    with expect_json_log() as json_log:
+        logger.critical("should log")
+
+    assert isinstance(json_log, JsonLog)
+    assert json_log.service == config.service_name
+    assert json_log.instance == config.service_instance_id
+    assert json_log.message == "should log"
+
+
+def test_record_compiler(caplog):
+    """Unit test for the RecordCompiler class."""
+    record_compiler = RecordCompiler(config=DEFAULT_CONFIG)
+
+    assert not caplog.records
+    log = logging.getLogger()  # getting root logger, not configured
+    log.setLevel("INFO")
+    log.error("This is a test")
+
+    assert len(caplog.records) == 1
+    record = caplog.records[0]
+
+    for key in ADDED_KEYS:
+        assert key not in record.__dict__
+
+    # Feed the record to the RecordCompiler
+    record_compiler.handle(record)
+    for key in DEFAULT_KEYS:
+        assert key in record.__dict__
+
+
+def test_json_formatter(caplog):
+    """Test that the JsonFormatter works like expected"""
+    formatter = JsonFormatter()
+
+    assert not caplog.records
+    log = logging.getLogger()  # just get root logger
+    log.setLevel("INFO")
+    log.error("This is a %s", "test")  # test with arg to make sure this isn't lost
+
+    assert len(caplog.records) == 1
+    record = caplog.records[0]
+
+    # we get an error if we skip the RecordCompiler
+    with pytest.raises(KeyError):
+        formatter.format(record)
+
+    # use the record compiler to stick in the extra fields
+    record_compiler = RecordCompiler(config=DEFAULT_CONFIG)
+    record_compiler.handle(record)
+
+    # format with the json formatter
+    output = formatter.format(record)
+
+    # load back into json format and verify the expected keys are all there
+    json_log = json.loads(output)
+    for key in DEFAULT_KEYS:
+        assert key in json_log
+
+    # make sure the message is what we expected
+    assert json_log["message"] == "This is a test"
+
+
+@pytest.mark.parametrize(
+    "log_format, formatter_class",
+    [
+        ("", JsonFormatter),
+        ("%(message)s", logging.Formatter),
+    ],
+)
+def test_formatter_selection(log_format: str, formatter_class: type[logging.Formatter]):
```

Comment:
```suggestion
def test_formatter_selection(log_format: Optional[str], formatter_class: type[logging.Formatter]):
```

Reply:
Changes in [95ad845](https://github.com/ghga-de/hexkit/pull/77/commits/95ad845d31aed82970ecc469a48c74ea70978a29)